### ISSUE TITLE
gomodifytags is broken in neovim v0.8.3

### ISSUE THREAD

**Issue by ArtAndreev** [jose-elias-alvarez/null-ls.nvim#1454]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

NVIM v0.8.3 Build type: Release LuaJIT 2.1.0-beta3 Compiled by brew@Ventura-arm64.local  Features: +acl +iconv +tui See ":help feature-compile"     system vimrc file: "$VIM/sysinit.vim"   fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.8.3/share/nvim"  Run :checkhealth for more info

### Dev Version?

- [X] I am using a stable Neovim release version, or if I am using a dev version of Neovim I have confirmed that my issue is reproducible on a stable version.

### Operating System

macOS Ventura 13.2.1

### Minimal Config

```Lua
-- this template is borrowed from nvim-lspconfig
local on_windows = vim.loop.os_uname().version:match("Windows")

local function join_paths(...)
    local path_sep = on_windows and "\\" or "/"
    local result = table.concat({ ... }, path_sep)
    return result
end

vim.g.loaded_remote_plugins = ""
vim.cmd([[set runtimepath=$VIMRUNTIME]])

local temp_dir = vim.loop.os_getenv("TEMP") or "/tmp"

vim.cmd("set packpath=" .. join_paths(temp_dir, "nvim", "site"))

local package_root = join_paths(temp_dir, "nvim", "site", "pack")
local install_path = join_paths(package_root, "packer", "start", "packer.nvim")
local compile_path = join_paths(install_path, "plugin", "packer_compiled.lua")

local null_ls_config = function()
    local null_ls = require("null-ls")
    -- add only what you need to reproduce your issue
    null_ls.setup({
        sources = {
            null_ls.builtins.code_actions.gomodifytags,
        },
        debug = true,
    })
end

local lspconfig_config = function()
    require('lspconfig').gopls.setup({
        on_attach = function(_, bufnr)
            local bufopts = {noremap = true, silent = true, buffer = bufnr}
            vim.keymap.set('n', '<leader>ca', vim.lsp.buf.code_action, bufopts)
        end,
    })
end

local treesitter_config = function()
    require('nvim-treesitter.configs').setup({
        ensure_installed = { 'go' },
        highlight = {
          enable = true,
        },
    })
end

local function load_plugins()
    -- only add other plugins if they are necessary to reproduce the issue
    require("packer").startup({
        {
            "wbthomason/packer.nvim",
            {
                'neovim/nvim-lspconfig',
                config = lspconfig_config,
            },
            -- Forces update of code actions.
            {
                'kosayoda/nvim-lightbulb',
                requires = 'antoinemadec/FixCursorHold.nvim',
            },
            {
                'nvim-treesitter/nvim-treesitter',
                run = function()
                    local ts_update = require('nvim-treesitter.install').update({ with_sync = true })
                    ts_update()
                end,
                config = treesitter_config,
            },
            {
                "jose-elias-alvarez/null-ls.nvim",
                after = 'nvim-treesitter',
                requires = { "nvim-lua/plenary.nvim" },
                config = null_ls_config,
            },
        },
        config = {
            package_root = package_root,
            compile_path = compile_path,
        },
    })
end

if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system({ "git", "clone", "https://github.com/wbthomason/packer.nvim", install_path })
    load_plugins()
    require("packer").sync()
else
    load_plugins()
    -- require("packer").sync()
end

vim.api.nvim_create_autocmd({'CursorHold', 'CursorHoldI'}, {
    group = vim.api.nvim_create_augroup('update_lightbulb', { clear = true }),
    desc = 'Update lightbulb for code actions',
    pattern = '*',
    callback = function()
        require('nvim-lightbulb').update_lightbulb()
    end,
})
```

### Steps to Reproduce

Run go project, wait for plugins and installation finish of go treesitter grammar, reopen go project, move cursor and get an error:

```
[null-ls] failed to run generator: ....nvim/lua/null-ls/builtins/code_actions/gomodifytags.lua:112: attempt to call field 'get_node' (a nil value)
```

Seems like person who authored https://github.com/jose-elias-alvarez/null-ls.nvim/pull/1443 used nightly version of neovim (v0.9.0) or read docs from master (site uses master version).

### Reproducibility Check

- [X] I confirm that my minimal config is based on the `minimal_init.lua` template and that my issue is reproducible by running `nvim --clean -u minimal_init.lua` and following the steps above.

### Expected Behavior

No error, `gomodifytags` works.

### Actual Behavior

Got an error.

### Debug Log

[TRACE Thu Mar 16 19:49:37 2023] ...site/pack/packer/opt/null-ls.nvim/lua/null-ls/client.lua:97: starting null-ls client
[TRACE Thu Mar 16 19:49:37 2023] ...im/site/pack/packer/opt/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method initialize
[DEBUG Thu Mar 16 19:49:37 2023] ...site/pack/packer/opt/null-ls.nvim/lua/null-ls/client.lua:152: unable to notify client for method textDocument/didOpen (client not active): {
  textDocument = {
    uri = "file:///private/var/folders/p8/jznrcvb54gl91s6fg50bznhm0000gn/T/go-try/main.go"
  }
}
[TRACE Thu Mar 16 19:49:37 2023] ...im/site/pack/packer/opt/null-ls.nvim/lua/null-ls/rpc.lua:127: received LSP notification for method initialized
[TRACE Thu Mar 16 19:49:37 2023] ...im/site/pack/packer/opt/null-ls.nvim/lua/null-ls/rpc.lua:127: received LSP notification for method textDocument/didOpen
[TRACE Thu Mar 16 19:49:37 2023] .../pack/packer/opt/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_DIAGNOSTICS_ON_OPEN
[DEBUG Thu Mar 16 19:49:37 2023] .../pack/packer/opt/null-ls.nvim/lua/null-ls/generators.lua:24: no generators available
[TRACE Thu Mar 16 19:49:44 2023] ...im/site/pack/packer/opt/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method textDocument/codeAction
[TRACE Thu Mar 16 19:49:44 2023] .../pack/packer/opt/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_CODE_ACTION
[WARN  Thu Mar 16 19:49:44 2023] .../pack/packer/opt/null-ls.nvim/lua/null-ls/generators.lua:94: failed to run generator: ....nvim/lua/null-ls/builtins/code_actions/gomodifytags.lua:112: attempt to call field 'get_node' (a nil value)
[TRACE Thu Mar 16 19:49:44 2023] ...ack/packer/opt/null-ls.nvim/lua/null-ls/code-actions.lua:35: received code actions from generators
[TRACE Thu Mar 16 19:49:44 2023] ...ack/packer/opt/null-ls.nvim/lua/null-ls/code-actions.lua:36: {}

### Help

Yes

### Implementation Help

revert pull request https://github.com/jose-elias-alvarez/null-ls.nvim/pull/1443

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by jose-elias-alvarez** [comment]
Yes, it is a mistake (and it's my fault for not investigating the PR more thoroughly). We should gate the method usage behind a version check. Mind putting in a PR?
